kitsune/bundles.py

```diff
@@ -28,7 +28,7 @@
             "sumo/js/ui.js",
             "sumo/js/analytics.js",
             "sumo/js/surveygizmo.js",
-            "sumo/js/instant_search.js",
+            "sumo/js/instant_search.es6",
```

Comment:
It gets a bit messy, since pipeline assumes `filename.split(".")[-1]` is the extension, and does its replacement on that part only.

So `instant_search.es6` works out of the box, because it gets transpiled into `instant_search.js`.

However, if we define our own compiler like:

```
from pipeline.compilers.es6 import ES6Compiler
class BabelCompiler(ES6Compiler):
    def match_file(self, path):
        return path.endswith('.es6.js')
```

Then `instant_search.es6.js` is transpiled into... `instant_search.es6.js`, so in dev the file just gets loaded from the app directly, and we don't run the transpiled JS.

Instead we have to define a compiler like:

```
class BabelCompiler(ES6Compiler):
    output_extension = "babelified.js"
    def match_file(self, path):
        return path.endswith('.es6.js')
```

So `instant_search.es6.js` is transpiled into `instant_search.es6.babelified.js` and then things work as they should in dev again (since the files are different names).

Since django-pipeline already has `.es6` as the default, though, I'm not confident overriding that behaviour.

Reply:
You can always use split(.) and look for es6. The reason for saying this is b/c of editors and the experience that contributors might have with this. You will need to tweak your editor to associate es6 => js. In any case, at least let's capture this in an issue and come back to it after the general release of ES7.